src/Runner/Runner.php

```diff
@@ -47,68 +63,216 @@ final class Runner
 
     private CacheManagerInterface $cacheManager;
 
-    private bool $isDryRun;
-
     private LinterInterface $linter;
 
     /**
-     * @var \Traversable<\SplFileInfo>
+     * @var ?iterable<\SplFileInfo>
      */
-    private $finder;
+    private $fileIterator;
+
+    private int $fileCount;
 
     /**
      * @var list<FixerInterface>
      */
     private array $fixers;
 
-    private bool $stopOnViolation;
-
     /**
-     * @param \Traversable<\SplFileInfo> $finder
-     * @param list<FixerInterface>       $fixers
+     * @param null|iterable<\SplFileInfo> $fileIterator
+     * @param list<FixerInterface>        $fixers
      */
     public function __construct(
-        \Traversable $finder,
+        RunnerConfig $runnerConfig,
+        ?iterable $fileIterator,
         array $fixers,
```

Comment:
other params are more like "injected services".
I wonder if `fileIterator` or `fixers` shouldn't become part of `RunnerConfig`

Reply:
It's not related to parallelisation and I don't plan to refactor it in this PR.

Reply:
tbh independently of class being internal or not I'm still interested in answer.
there were not `RunnerConfig` so creating `RunnerConfig` and defining what's inside is scope of the PR ;)

[but yeah, let's solve `@internal` first]

Reply:
I created `RunnerConfig` as a holder for configurable params that are immutable across the single execution. That's why the file iterator is not there because it's a moving part in the worker. However a collection of fixers could be part of that class, since it's immutable for each analysed file/chunk.

Reply:
I'll revert previous signature, we can refactor it in the future. Added todo item in issue's description, so I'm going to resolve this discussion so it's easier to track what's left.